gslib/commands/publicaccessprevention.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module provides the publicaccessprevention command to gsutil."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import textwrap
+
+from gslib import metrics
+from gslib.command import Command
+from gslib.command_argument import CommandArgument
+from gslib.cs_api_map import ApiSelector
+from gslib.exception import CommandException
+from gslib.exception import NO_URLS_MATCHED_TARGET
+from gslib.help_provider import CreateHelpText
+from gslib.third_party.storage_apitools import storage_v1_messages as apitools_messages
+from gslib.utils.constants import NO_MAX
+
+_SET_SYNOPSIS = """
+  gsutil publicaccessprevention set (enforced|unspecified) gs://<bucket_name>...
+"""
+
+_GET_SYNOPSIS = """
+  gsutil publicaccessprevention get bucket_url...
+"""
+
+_SYNOPSIS = _SET_SYNOPSIS + _GET_SYNOPSIS.lstrip('\n')
+
+_SET_DESCRIPTION = """
+<B>SET</B>
+  The ``publicaccessprevention set`` command configures public access prevention
+  for Google Cloud Storage buckets.
+
+<B>SET EXAMPLES</B>
+  Configure your buckets to use public access prevention:
+
+    gsutil publicaccessprevention set enforced gs://redbucket gs://bluebucket
+
+  Configure your buckets to NOT use public access prevention:
+
+    gsutil publicaccessprevention set unspecified gs://redbucket gs://bluebucket
+"""
+
+_GET_DESCRIPTION = """
+<B>GET</B>
+  The ``publicaccessprevention get`` command returns public access prevention
+  value for the specified Cloud Storage bucket(s).
```

Comment:
Just "buckets" here is fine - we try not to use the (s) syntax.

Reply:
I count 10 uses of `bucket(s)` in this package. Perhaps we should do a pass on the existing descriptions.